### PR TITLE
internal/import-files-to-item

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  5  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

**Review by vidiecan — CHANGES_REQUESTED** [review]
you added 100 equal files, like really?!

File: data_pump/utils.py [hunk 1]
```diff
@@ -94,3 +95,23 @@ def save_dict_as_json(json_name, dictionary: dict):
 def create_dict_from_json(json_name):
     return {int(key): value
             for key, value in read_json(json_name, MAPPING_PATH).items()}
+
+
+def create_bitstream(bundle=None, name=None, path=None, mime=None, metadata=None):
```
Comment (vidiecan) [inline data_pump/utils.py:100]: these specific functions should not be here

File: support/dspace_interface/client.py [hunk 2]
```diff
@@ -671,9 +671,21 @@ def create_bitstream(self, bundle=None, name=None, path=None, mime=None,
         payload = {'properties': json.dumps(properties) + ';application/json'}
         h = self.session.headers
         h.update({'Content-Encoding': 'gzip'})
-        req = Request('POST', url, data=payload, headers=h, files=files)
-        prepared_req = self.session.prepare_request(req)
-        r = self.session.send(prepared_req)
+
+        mp_encoder = MultipartEncoder(
+            fields={
+                'file': (name, open(path, 'rb'), 'text/plain'),
```
Comment (vidiecan) [inline support/dspace_interface/client.py:677]: encoding is missing, you sure text/plain is ok?

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: data_pump/utils.py (hunk 1 above)
Comment (milanmajchrak) [reply data_pump/utils.py:100]: Why? It should be in another place or it should be removed..?

**Comment by milanmajchrak** [comment]
> you added 100 equal files, like really?!

Yes, you asked me to create an item and import a huge bunch of files. The assignment was clear. What do you want to have imported there?

File: support/dspace_interface/client.py (hunk 2 above)
Comment (milanmajchrak) [reply support/dspace_interface/client.py:677]: Encoding info is not needed there. I removed that `text/plain`. My mistake.

File: data_pump/utils.py (hunk 1 above)
Comment (milanmajchrak) [reply data_pump/utils.py]: Yes, you are right. I realized that I made this feature from the non up to date `main` branch. I didn't have pulled `main` branch... but now it is up to date.

**Review by vidiecan — CHANGES_REQUESTED** [review]
summary of our talk:
1. add 1 file to git, import it 100x
2. is src/dspace/_rest.py really the place for dspace.object specific rest calls?

**Review by MajoBerger — APPROVED** [review]
untested, but I see no obvious flaw

**Pull request merged**